### ISSUE TITLE
Export public and private key -- can't import -- Cannot read property 'userid' of null

### ISSUE THREAD

**Issue by xxdesmus** [mailvelope/mailvelope#496]
It would seem your export functionality is broken in v1.7.1

I exported a public and private key -- and then went to a new computer and tried to import it and that fails.

`Import Error. Unable to import a key due to exception: TypeError: Cannot read property 'userid' of null`

I can't imagine I'm the only one with this issue so I figured I'd report this. 

This is obviously a pretty critical if confirmed -- PGP in a browser becomes pretty useless if you're locked to a single laptop/browser.

Steps to reproduce: export -> all -> transfer file to new laptop also running Chrome extension v1.7.1 --> import and then you receive the error when trying to import the private key. The public key imports fine.

**Comment by toberndo** [comment]
I cannot reproduce with any key that I have. Looks like we introduced a bug on key import for keys that could be imported in previous releases. Is the key functional (for e.g. decrypt) in Mailvelope? Could you by any chance send the output of `gpg --list-packets your_key.asc` on the file of the private key?

**Comment by xxdesmus** [comment]
Hi! The key is functional on 2 other laptops -- has been functional for roughly 1 year on both laptops. I just received a new laptop and I'm trying to migrate to this new laptop when I encountered the issue.

Perhaps this is relevant. When I run the `--list-packets` against my key directly at the command line with gpg I get:

```
GPG -->

:secret key packet:
    version 4, algo 1, created 1373057649, expires 0
    skey[0]: [4096 bits]
    skey[1]: [17 bits]
    iter+salt S2K, algo: 3, SHA1 protection, hash: 8, salt: XXXXXXXXXX
    protect count: 65536 (96)
    protect IV:  db 4c 72 1d 51 1d 95 99
    encrypted stuff follows
    keyid: 57B60114DE0B314D
:user ID packet: "Justin XXXXXXXXXX <justin@XXXXXXXXXX>"
:signature packet: algo 1, keyid 57B60114DE0B314D
    version 4, created 1373059569, md5len 0, sigclass 0x13
    digest algo 10, begin of digest 72 f7
    hashed subpkt 2 len 4 (sig created 2013-07-05)
    hashed subpkt 27 len 1 (key flags: 2F)
    hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
    hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
    hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
    hashed subpkt 30 len 1 (features: 01)
    hashed subpkt 23 len 1 (key server preferences: 80)
    subpkt 16 len 8 (issuer key ID 57B60114DE0B314D)
    data: [4095 bits]
```

vs. what Mailvelope exports, and then I run `--list-packets` against that key -->

```
Mailvelope —>

:secret key packet:
    version 4, algo 1, created 1373057649, expires 0
    skey[0]: [4096 bits]
    skey[1]: [17 bits]
    iter+salt S2K, algo: 3, SHA1 protection, hash: 8, salt: XXXXXXXXXX
    protect count: 65536 (96)
    protect IV:  db 4c 72 1d 51 1d 95 99
    encrypted stuff follows
    keyid: 57B60114DE0B314D
:user ID packet: "Justin XXXXXXXXXX <justin@XXXXXXXXXX>"
:signature packet: algo 1, keyid 57B60114DE0B314D
    version 4, created 1373059569, md5len 0, sigclass 0x13
    digest algo 10, begin of digest 72 f7
    hashed subpkt 2 len 4 (sig created 2013-07-05)
    hashed subpkt 27 len 1 (key flags: 2F)
    hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
    hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
    hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
    hashed subpkt 30 len 1 (features: 01)
    hashed subpkt 23 len 1 (key server preferences: 80)
    subpkt 16 len 8 (issuer key ID 57B60114DE0B314D)
    data: [4095 bits]
:signature packet: algo 1, keyid 57B60114DE0B314D
    version 4, created 1398276336, md5len 0, sigclass 0x13
    digest algo 10, begin of digest 6e 56
    hashed subpkt 27 len 1 (key flags: 2F)
    hashed subpkt 30 len 1 (features: 01)
    hashed subpkt 23 len 1 (key server preferences: 80)
    hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
    hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
    hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
    hashed subpkt 2 len 4 (sig created 2014-04-23)
    hashed subpkt 25 len 1 (primary user ID)
    subpkt 16 len 8 (issuer key ID 57B60114DE0B314D)
    data: [4095 bits]
:signature packet: algo 1, keyid 57B60114DE0B314D
    version 4, created 1395796806, md5len 0, sigclass 0x13
    digest algo 10, begin of digest 6e 24
    hashed subpkt 27 len 1 (key flags: 2F)
    hashed subpkt 30 len 1 (features: 01)
    hashed subpkt 23 len 1 (key server preferences: 80)
    hashed subpkt 2 len 4 (sig created 2014-03-26)
    hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
    hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
    hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
    subpkt 16 len 8 (issuer key ID 57B60114DE0B314D)
    data: [4094 bits]
:signature packet: algo 1, keyid 57B60114DE0B314D
    version 4, created 1397500538, md5len 0, sigclass 0x13
    digest algo 10, begin of digest 8d 06
    hashed subpkt 27 len 1 (key flags: 2F)
    hashed subpkt 30 len 1 (features: 01)
    hashed subpkt 23 len 1 (key server preferences: 80)
    hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
    hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
    hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
    hashed subpkt 2 len 4 (sig created 2014-04-14)
    hashed subpkt 25 len 1 (primary user ID)
    subpkt 16 len 8 (issuer key ID 57B60114DE0B314D)
    data: [4095 bits]
:signature packet: algo 1, keyid 14CADC031BA1A8DF
    version 4, created 1397781468, md5len 0, sigclass 0x10
    digest algo 2, begin of digest c0 1d
    hashed subpkt 2 len 4 (sig created 2014-04-18)
    subpkt 16 len 8 (issuer key ID 14CADC031BA1A8DF)
    data: [4096 bits]
```

Thanks for the help!

**Comment by aleksandar-todorovic** [comment]
I've encountered the same error with the key that could be found in the following address: https://keybase.io/pudo/pgp_keys.asc

Notice how this is _not_ exported from Mailvelope and only contains the public key (associated with multiple email addresses).

To make this issue even funkier, I've installed Mailvelope _twice_ in the last hour or so. I've hit the error on the first time, and, for the life of it, I can't replicate the issue after the reinstall on the same browser.

I genuinely can't figure out how to replicate this.

At the very first time, I've tried the following:

* Downloading the key from Keybase.
* Exporting the key from Thunderbird + Enigmail.
* Downloading from the keyserver.
* Exporting using `gpg --export -a "identifier" -> identifier.asc`

Every single time, I've hit the same road block:

```
Import Error. Unable to import a key due to exception: TypeError: Cannot read property 'userid' of null
```

So, I wanted to start from a clean state, and deleting Mailvelope was simpler than deleting all the public keys individually, and when I reinstalled Mailvelope, I've never hit the same road block again and was successfully able to import the key using any of the methods I've mentioned above.

My googling around led me to this issue and the issue #376. They're both about exporting public and private keys from Mailvelope, and none of them is related to importing a public key associated with multiple email addresses.

Now, obviously, I'm not able to say to the person experiencing this to simply reinstall Mailvelope, since that would require exporting and importing all of the keys and I have no clue if that process will trigger the same error, so I'm genuinely not able to figure out how to proceed from here.

`gpg --list-packets` output:

```
# off=0 ctb=99 tag=6 hlen=3 plen=525
:public key packet:
	version 4, algo 1, created 1418557461, expires 0
	pkey[0]: [4096 bits]
	pkey[1]: [17 bits]
	keyid: 6B33881D2DA6741F
# off=528 ctb=b4 tag=13 hlen=2 plen=38
:user ID packet: "Friedrich Lindenberg <pudo@keybase.io>"
# off=568 ctb=89 tag=2 hlen=3 plen=573
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1418557645, md5len 0, sigclass 0x13
	digest algo 10, begin of digest 78 91
	hashed subpkt 2 len 4 (sig created 2014-12-14)
	hashed subpkt 27 len 1 (key flags: 03)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
	hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
	hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
	hashed subpkt 30 len 1 (features: 01)
	hashed subpkt 23 len 1 (keyserver preferences: 80)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4095 bits]
# off=1144 ctb=b4 tag=13 hlen=2 plen=39
:user ID packet: "Friedrich Lindenberg <pudo@fastmail.fm>"
# off=1185 ctb=89 tag=2 hlen=3 plen=573
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1418562462, md5len 0, sigclass 0x13
	digest algo 10, begin of digest 66 cd
	hashed subpkt 2 len 4 (sig created 2014-12-14)
	hashed subpkt 27 len 1 (key flags: 03)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
	hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
	hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
	hashed subpkt 30 len 1 (features: 01)
	hashed subpkt 23 len 1 (keyserver preferences: 80)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4093 bits]
# off=1761 ctb=b4 tag=13 hlen=2 plen=41
:user ID packet: "Friedrich Lindenberg <friedrich@pudo.org>"
# off=1804 ctb=89 tag=2 hlen=3 plen=576
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1418557588, md5len 0, sigclass 0x13
	digest algo 10, begin of digest 15 74
	hashed subpkt 27 len 1 (key flags: 03)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
	hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
	hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
	hashed subpkt 30 len 1 (features: 01)
	hashed subpkt 23 len 1 (keyserver preferences: 80)
	hashed subpkt 2 len 4 (sig created 2014-12-14)
	hashed subpkt 25 len 1 (primary user ID)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4095 bits]
# off=2383 ctb=b4 tag=13 hlen=2 plen=43
:user ID packet: "Friedrich Lindenberg <flindenberg@icfj.org>"
# off=2428 ctb=89 tag=2 hlen=3 plen=573
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1418562396, md5len 0, sigclass 0x13
	digest algo 10, begin of digest 12 51
	hashed subpkt 2 len 4 (sig created 2014-12-14)
	hashed subpkt 27 len 1 (key flags: 03)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
	hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
	hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
	hashed subpkt 30 len 1 (features: 01)
	hashed subpkt 23 len 1 (keyserver preferences: 80)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4089 bits]
# off=3004 ctb=b4 tag=13 hlen=2 plen=52
:user ID packet: "Friedrich Lindenberg <friedrich.lindenberg@okfn.org>"
# off=3058 ctb=89 tag=2 hlen=3 plen=573
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1418562415, md5len 0, sigclass 0x13
	digest algo 10, begin of digest ab 78
	hashed subpkt 2 len 4 (sig created 2014-12-14)
	hashed subpkt 27 len 1 (key flags: 03)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
	hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
	hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
	hashed subpkt 30 len 1 (features: 01)
	hashed subpkt 23 len 1 (keyserver preferences: 80)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4096 bits]
# off=3634 ctb=b4 tag=13 hlen=2 plen=53
:user ID packet: "Friedrich Lindenberg <friedrich.lindenberg@gmail.com>"
# off=3689 ctb=89 tag=2 hlen=3 plen=573
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1418557562, md5len 0, sigclass 0x13
	digest algo 10, begin of digest 76 28
	hashed subpkt 2 len 4 (sig created 2014-12-14)
	hashed subpkt 27 len 1 (key flags: 03)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
	hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
	hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
	hashed subpkt 30 len 1 (features: 01)
	hashed subpkt 23 len 1 (keyserver preferences: 80)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4095 bits]
# off=4265 ctb=d1 tag=17 hlen=6 plen=11950 new-ctb
:attribute packet: [jpeg image of size 11928]
# off=16221 ctb=89 tag=2 hlen=3 plen=573
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1418557870, md5len 0, sigclass 0x13
	digest algo 10, begin of digest 19 a8
	hashed subpkt 2 len 4 (sig created 2014-12-14)
	hashed subpkt 27 len 1 (key flags: 03)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
	hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
	hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
	hashed subpkt 30 len 1 (features: 01)
	hashed subpkt 23 len 1 (keyserver preferences: 80)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4096 bits]
# off=16797 ctb=b4 tag=13 hlen=2 plen=37
:user ID packet: "Friedrich Lindenberg <pudo@occrp.org>"
# off=16836 ctb=89 tag=2 hlen=3 plen=573
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1447591153, md5len 0, sigclass 0x13
	digest algo 10, begin of digest 93 7e
	hashed subpkt 2 len 4 (sig created 2015-11-15)
	hashed subpkt 27 len 1 (key flags: 03)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	hashed subpkt 11 len 4 (pref-sym-algos: 9 8 7 3)
	hashed subpkt 21 len 4 (pref-hash-algos: 10 9 8 11)
	hashed subpkt 22 len 4 (pref-zip-algos: 2 3 1 0)
	hashed subpkt 30 len 1 (features: 01)
	hashed subpkt 23 len 1 (keyserver preferences: 80)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4093 bits]
# off=17412 ctb=b9 tag=14 hlen=3 plen=525
:public sub key packet:
	version 4, algo 1, created 1418557461, expires 0
	pkey[0]: [4096 bits]
	pkey[1]: [17 bits]
	keyid: 145F820FFCEFA2C8
# off=17940 ctb=89 tag=2 hlen=3 plen=549
:signature packet: algo 1, keyid 6B33881D2DA6741F
	version 4, created 1418557461, md5len 0, sigclass 0x18
	digest algo 10, begin of digest d4 17
	hashed subpkt 2 len 4 (sig created 2014-12-14)
	hashed subpkt 27 len 1 (key flags: 0C)
	hashed subpkt 9 len 4 (key expires after 4y1d0h0m)
	subpkt 16 len 8 (issuer key ID 6B33881D2DA6741F)
	data: [4096 bits]
```

This is the second time I'm dealing with this error since this issue got created (even though it's a duplicate from #376 which is over a year old). Both of the times, the key has been issued for multiple email addresses.

**Comment by alberttwong** [comment]
I also have this same error.  I'm trying to import my secret key from PGP into mailvelope.  I can send logs if needed.

**Comment by alberttwong** [comment]
I had to delete the plugin and then delete all the entries in "display keys" and then import and it imported my primary key correctly.

**Comment by aleksandar-todorovic** [comment]
Two freaking years have passed and the only comment from @mailvelope is @toberndo's "unable to reproduce" on the first day?

This has got to be a joke.

**Comment by alberttwong** [comment]
@aleksandar-todorovic This is probably an open source project done in his/her free time.   Engineers need to be paid for their time.    We can't expect people to give support and fixes relying on their good will.  If we want it fixed, offer them money.

**Comment by toberndo** [comment]
This should be fixed with Mailvelope 3.0 released Dec 11th. Please check with newest release.
